torchtitan/models/llama/model.py

```diff
@@ -321,14 +371,20 @@ def forward(
 
         """
         h = x + self.attention(self.attention_norm(x), freqs_cis)
-        out = h + self.feed_forward(self.ffn_norm(h))
+        if not self.enable_moe:
```

Comment:
wrt to above, this is what I mean about implying it's dense but not really making it clear....is_dense_model is more precise.

Reply:
I modified it to"
~~~
if self.is_dense_model:
     out = h + self.feed_forward(self.ffn_norm(h))
elif self.is_moe_model:
     out = h + self.moe(self.ffn_norm(h))
else:
     raise NotImplementedError("unknown model type")
~~~
in my copy of your PR..mostly showing how it reads with the is_blah_model approach.